merkle/log_verifier.go

```diff
@@ -212,3 +213,34 @@ func (v LogVerifier) VerifyConsistencyProof(snapshot1, snapshot2 int64, root1, r
 
 	return nil // Proof OK.
 }
+
+// PrefixHashFromInclusionProof calculates a root hash over leaves [0..index),
+// based on the inclusion |proof| and |leafHash| for a leaf at the specified
+// |index| in a tree of the specified |size| with the passed in |root| hash.
+// Accepts 0 <= index < size.
+// Returns an error if the proof verification fails.
+func (v LogVerifier) PrefixHashFromInclusionProof(
```

Comment:
The name should probably include Verify for consistency. Or take the verify step out and make it the callers responsibility.